### PR TITLE
fix: Migration for changing gender type

### LINKED ISSUE #6106: Write migration file for gender in attendee custom form from text to select
Referrence : https://github.com/fossasia/open-event-frontend/pull/3180

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #6106 

#### Short description of what this resolves:

- Migrating `gender field` type from `text` to `select`

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by kushthedude** [comment]
@uds5501 @shreyanshdwivedi Since this is a manual migration file, Can you please for any potential issue which can be created by it?

File: migrations/versions/43e8c59339af_.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import sqlalchemy_utils
+
+
+ # revision identifiers, used by Alembic.
+revision = '6f7b6fad3f54'
+down_revision = '6f7b6fad3f53'
+
+
+def upgrade():
+    op.execute("UPDATE custom_forms SET type = 'select' where field_identifier = 'gender';", execution_options=None)
```
Comment (CosmicCoder96) [inline migrations/versions/43e8c59339af_.py]: make it more specific, only the form type 'attendee' is being refactored.

Comment (kushthedude) [reply migrations/versions/43e8c59339af_.py]: @CosmicCoder96 How do I select the `form_type` to `attendee`, Will adding an `and form = 'attendee'` be suffice or there is some other way?

**Comment by kushthedude** [comment]
@mrsaicharan1 @shreyanshdwivedi @uds5501 Please Review

File: migrations/versions/43e8c59339af_.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+down_revision = 'b08a4ffff5dd'
+
+
+def upgrade():
+    op.execute("UPDATE custom_forms SET type = 'select' where field_identifier = 'gender' and form = 'attendee';", execution_options=None)
+
+
+
+def downgrade():
+    op.execute("UPDATE custom_forms SET type = 'text' where field_identifier = 'gender';", execution_options=None)
```
Comment (shreyanshdwivedi) [inline migrations/versions/43e8c59339af_.py]: @kushthedude don't you think you should add `form =attendee` here too?

Comment (kushthedude) [reply migrations/versions/43e8c59339af_.py]: @shreyanshdwivedi Updated It

**Comment by kushthedude** [comment]
@iamareebjamal Please Review

**Comment by kushthedude** [comment]
@iamareebjamal @mrsaicharan1 @CosmicCoder96 @uds5501 Please Review

**Comment by kushthedude** [comment]
@iamareebjamal @mrsaicharan1 @CosmicCoder96 @uds5501 Please Review

**Pull request merged**